packages/x-components/src/x-modules/facets/components/facets/facets-provider.vue

```diff
@@ -99,7 +99,8 @@
     protected extractSelectedFilters(facets: Facet[]): void {
       this.selectedFilters = facets
         .flatMap(facet => facet.filters)
-        .filter(filter => filter.selected);
+        .filter(filter => filter.selected)
+        .map(filter => ({ ...filter }));
```

Comment:
not necessary, we only need the filters to compare, and for that, it is only necessary the id.

We could only store the id: `.map(filter => {id: filter.id})` and it will keep working

Reply:
Allright for the first part. You are right, we only need the id.

But for the second part, the component fails if we have hierarchical filters selected, so I still think we should add the hierarchical filters here.

```ts
        .flatMap(facet =>
          isHierarchicalFacet(facet) ? flatHierarchicalFilters(facet.filters) : facet.filters
        )
```

Reply:
done